core/log.c

```diff
@@ -148,32 +148,26 @@ glvl_allowed(register GLogLevelFlags lvl)
 		|| (ALLOWED_LEVEL() >= REAL_LEVEL(lvl));
 }
 
-static void
-_purify(register gchar *s)
-{
-	static guint8 invalid[256] = {0};
-	if (!invalid[0]) {
-		for (int i=0; i<256 ;i++)
-			invalid[i] = g_ascii_isspace(i) || !g_ascii_isprint(i);
-	}
+static guint8 map_valid_ascii[256] = {0};
+
+void _constructor_map_valid_ascii (void);
 
-	for (gchar c; (c=*s) ; s++) {
-		if (invalid[(guint8)c])
-			*s = ' ';
+void __attribute__ ((constructor))
+_constructor_map_valid_ascii (void)
+{
+	/* 0 remains 0, and the array is already zero'ed */
+	for (int i=1; i<256 ;i++) {
+		map_valid_ascii[i] = (g_ascii_isspace(i) || !g_ascii_isprint(i))
+			? (guint8)' ' : (guint8)i;
```

Comment:
New lines are converted to spaces. Do we still want that?

Reply:
I do, but I am not the majority, even in terms of weight.